src/infercnvpy/tl/_infercnv.py

```diff
@@ -161,26 +181,83 @@ def _running_mean(x: np.ndarray | scipy.sparse.spmatrix, n: int = 50, step: int
         Length of the running window
     step
         only compute running windows ever `step` columns, e.g. if step is 10
-        0:100, 10:110, 20:120 etc. Saves memory.
+        0:99, 10:109, 20:119 etc. Saves memory.
     """
     if n < x.shape[1]:  # regular convolution: the filter is smaller than the #genes
         r = np.arange(1, n + 1)
-
         pyramid = np.minimum(r, r[::-1])
-        smoothed_x = np.apply_along_axis(lambda row: np.convolve(row, pyramid, mode="same"), axis=1, arr=x) / np.sum(
-            pyramid
-        )
-        return smoothed_x[:, np.arange(0, smoothed_x.shape[1], step)]
+        smoothed_x = np.apply_along_axis(
+            lambda row: np.convolve(row, pyramid, mode="valid"),
+            axis=1,
+            arr=x,
+        ) / np.sum(pyramid)
 
-    else:  # there's less genes than the filtersize. just apply a single conv with a smaller filter (no sliding)
-        r = np.arange(1, x.shape[1] + 1)
-        pyramid = np.minimum(r, r[::-1])
-        smoothed_x = x @ pyramid.reshape(-1, 1)
-        smoothed_x = smoothed_x / pyramid.sum()
-        return smoothed_x
+        ## get the indices of the genes used in the convolution
+        convolution_indices = get_convolution_indices(x, n)[np.arange(0, smoothed_x.shape[1], step)]
```

Comment:
Can the `step` be integrated into get_convolution_indices directly? Currently, a large array of indices is generated, but only a fraction of it is ever used (because of `step`).